connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/table/BigtableTableTest.java

```diff
@@ -0,0 +1,252 @@
+                    Column.physical(
+                            TestingUtils.NESTED_COLUMN_FAMILY_2,
+                            DataTypes.ROW(
+                                    DataTypes.FIELD(
+                                            TestingUtils.STRING_FIELD_2, DataTypes.STRING()))));
+
+    @Test
+    public void testDynamicTableBigtableSink() throws IOException {
+        Map<String, String> options = getRequiredOptions();
+        options.put(BigtableConnectorOptions.COLUMN_FAMILY.key(), TestingUtils.COLUMN_FAMILY);
+
+        ResolvedSchema schema = getResolvedSchema(false);
+
+        BigtableDynamicTableSink sink =
+                (BigtableDynamicTableSink) FactoryMocks.createTableSink(schema, options);
+
+        ReadableConfig connectorOptions = sink.connectorOptions;
+        assertEquals(
+                connectorOptions.get(BigtableConnectorOptions.COLUMN_FAMILY),
+                TestingUtils.COLUMN_FAMILY);
+        assertFalse(connectorOptions.get(BigtableConnectorOptions.USE_NESTED_ROWS_MODE));
+        assertEquals(connectorOptions.get(BigtableConnectorOptions.TABLE), TestingUtils.TABLE);
+        assertEquals(
+                connectorOptions.get(BigtableConnectorOptions.INSTANCE), TestingUtils.INSTANCE);
+        assertEquals(connectorOptions.get(BigtableConnectorOptions.PROJECT), TestingUtils.PROJECT);
+
+        assertEquals(sink.rowKeyField, TestingUtils.ROW_KEY_FIELD);
+        assertEquals(sink.resolvedSchema, schema);
+        assertNull(sink.parallelism);
+    }
+
+    @Test
+    public void testDynamicTableBigtableSinkParallelism() throws IOException {
+        Map<String, String> options = getRequiredOptions();
+        options.put(BigtableConnectorOptions.COLUMN_FAMILY.key(), TestingUtils.COLUMN_FAMILY);
+        options.put(BigtableConnectorOptions.SINK_PARALLELISM.key(), "2");
+
+        ResolvedSchema schema = getResolvedSchema(false);
+
+        BigtableDynamicTableSink sink =
+                (BigtableDynamicTableSink) FactoryMocks.createTableSink(schema, options);
+
+        ReadableConfig connectorOptions = sink.connectorOptions;
+        assertEquals(
+                connectorOptions.get(BigtableConnectorOptions.COLUMN_FAMILY),
+                TestingUtils.COLUMN_FAMILY);
+        assertFalse(connectorOptions.get(BigtableConnectorOptions.USE_NESTED_ROWS_MODE));
+        assertEquals(connectorOptions.get(BigtableConnectorOptions.TABLE), TestingUtils.TABLE);
+        assertEquals(
+                connectorOptions.get(BigtableConnectorOptions.INSTANCE), TestingUtils.INSTANCE);
+        assertEquals(connectorOptions.get(BigtableConnectorOptions.PROJECT), TestingUtils.PROJECT);
+
+        assertEquals(sink.rowKeyField, TestingUtils.ROW_KEY_FIELD);
+        assertEquals((Integer) sink.parallelism, (Integer) 2);
+        assertEquals(sink.resolvedSchema, schema);
+    }
+
+    @Test
+    public void testDynamicTableBigtableSinkNested() throws IOException {
```

Comment:
let's add a test to validate the behavior of USE_NESTED_ROWS_MODE but the schema is not a nested rows schema

Reply:
I should ahve add this in the PR, sorry:

All the schema checks are happening when the serializer is created, so we don't need to do them in the Table Factory